### ISSUE TITLE
Problem when use [EntityIndex] in a componet with namespace.

### ISSUE THREAD

**Issue by zjtzyrcfd** [sschmid/Entitas-CSharp#330]
generate with the following component
```

namespace Skill
{
    [Skill]
    public class OwnedBy : IComponent {
        [Entitas.CodeGenerator.Api.EntityIndex]
        public System.UInt64 ownerID;
    }
}
```
then in the generated file ContextsEntityIndices.cs:
```
//------------------------------------------------------------------------------
// <auto-generated>
//     This code was generated by Entitas.CodeGenerator.EntityIndexGenerator.
//
//     Changes to this file may cause incorrect behavior and will be lost if
//     the code is regenerated.
// </auto-generated>
//------------------------------------------------------------------------------
public partial class Contexts {

    public const string SkillOwnedBy = "SkillOwnedBy";

    [Entitas.CodeGenerator.Api.PostConstructor]
    public void InitializeEntityIndices() {
        game.AddEntityIndex(SkillOwnedBy, new Entitas.EntityIndex<GameEntity, ulong>(
            game.GetGroup(GameMatcher.SkillOwnedBy),
            (e, c) => { var component = c as Skill.OwnedBy; return component != null ? component.ownerID : e.ownedBy.ownerID; }));
    }
}
```
since the component is actually referenced by e.skillOwnedBy, this code reported an error.

**Comment by awalltoo** [comment]
On a related note, if you are only using the data provider for components without namespaces (i.e. not the "with namespaces" one), the generated code still tries to use the namespace version of the matcher name, which doesn't compile.  In the above example, it would refer to "GameMatcher.SkillOwnedBy" even though the correct way to refer to that matcher would be "GameMatcher.OwnedBy".

**Comment by rvangijlswijk** [comment]
I came across the same problem when using a component in a namespace. As mentioned, the `GetGroup()` call calls the component from the matcher and includes the namespace in the referenced component. This is incorrect, as the matcher doesn't know anything about namespaces. 

I would expect the call in the exampe above to be the following:
```
game.GetGroup(GameMatcher.OwnedBy),
```
This example above also fixes the compilation errors.

Moreover, and this is more of an enhancement, methods of components within a namespace can have a very long namespace. In the example above, the call would be `GetEntitiesWithSkillOwnedBy()`, which is still doable. However, when one has a full-blown namespace, the GetEntities method can get a really long name. E.g., when the namespace is <company>.<project>.ui, then the method name becomes:
```
GetEntitiesWithCompanyProjectUiOwnedBy()
```
... which is a bit too long, if you ask me. I would propose getting rid of of the namespace in the method call (or let the user choose in the preferences).

**Comment by rvangijlswijk** [comment]
Actually, I think this is a duplicate of #328 .

**Comment by sschmid** [comment]
Fixed in [0.39.0](https://github.com/sschmid/Entitas-CSharp/releases/tag/0.39.0)
